portal-impl/src/com/liferay/portal/service/persistence/impl/OrganizationFinderImpl.java

```diff
@@ -576,6 +591,94 @@ protected int countByOrganizationId(
 		return 0;
 	}
 
+	protected List<Object> doFindO_U_ByC_P(
+		long companyId, long parentOrganizationId,
+		QueryDefinition<?> queryDefinition) {
+
+		Session session = null;
+
+		try {
+			session = openSession();
+
+			StringBundler sb = new StringBundler(5);
+
+			sb.append(StringPool.OPEN_PARENTHESIS);
+
+			String sql = CustomSQLUtil.get(FIND_BY_C_P);
+
+			sql = StringUtil.replace(
+				sql, "SELECT organizationId",
+				"SELECT organizationId, 0 AS userId, name");
+			sql = StringUtil.replace(
+				sql, "(organizationId > ?) AND", StringPool.BLANK);
+
+			sb.append(sql);
+			sb.append(") UNION ALL (");
+
+			sql = CustomSQLUtil.get(FIND_U_BY_C_S_O);
```

Comment:
Although it's not uncommon to find this in Liferay code, reusing variables this way is a practice that can quickly lead to bugs.

My recommendation to solve this and at the same time increase the legibility of this code is to create two methods, getUsersSQL and getOrganizationsSQL, which encapsulate the sql manipulation logic for each and then are appended with the UNION ALL. See JournalFolderFinderImpl for an example of doing this.